src/hotspot/share/gc/g1/g1EvacFailure.cpp

```diff
@@ -205,12 +205,15 @@ class RemoveSelfForwardPtrHRClosure: public HeapRegionClosure {
   G1RedirtyCardsLocalQueueSet _rdc_local_qset;
   UpdateLogBuffersDeferred _log_buffer_cl;
 
+  uint volatile* _num_failed_regions;
+
 public:
-  RemoveSelfForwardPtrHRClosure(G1RedirtyCardsQueueSet* rdcqs, uint worker_id) :
+  RemoveSelfForwardPtrHRClosure(G1RedirtyCardsQueueSet* rdcqs, uint worker_id, uint volatile* num_failed_regions) :
```

Comment:
Maybe increment failed regions through G1CollectedHeap instead of a pointer to the value?

Reply:
You probably mean `G1ParRemoveSelfForwardPtrsTask`. I would not want to add that counter to `G1CollectedHeap` if possible. The suggestion seems to make the code just more verbose, not "cleaner", but that is subjective of course.

Reply:
Okay.